src/Ethereum/Signer.cpp

```diff
@@ -50,26 +47,30 @@ std::string Signer::signJSON(const std::string& json, const Data& key) {
     return hex(output.encoded());
 }
 
-Signature Signer::valuesRSV(const uint256_t& chainID, const Data& signature) noexcept {
+Signature Signer::signatureDataToStruct(const Data& signature) noexcept {
     boost::multiprecision::uint256_t r, s, v;
     import_bits(r, signature.begin(), signature.begin() + 32);
     import_bits(s, signature.begin() + 32, signature.begin() + 64);
     import_bits(v, signature.begin() + 64, signature.begin() + 65);
-    v += 27;
+    return Signature{r, s, v};
+}
 
-    boost::multiprecision::uint256_t newV;
+Signature Signer::signatureDataToStructWithEip155(const uint256_t& chainID, const Data& signature) noexcept {
+    Signature rsv = signatureDataToStruct(signature);
     if (chainID != 0) {
-        import_bits(newV, signature.begin() + 64, signature.begin() + 65);
-        newV += 35 + chainID + chainID;
+        rsv.v += 35 + chainID + chainID;
```

Comment:
can you leave comment for the future developers why it's adding here

Reply:
Done